Composer/packages/client/src/pages/botProject/table-view.tsx

```diff
@@ -0,0 +1,736 @@
+import { jsx } from '@emotion/core';
+import { useRecoilValue } from 'recoil';
+import React, { useMemo, useState, useEffect, useCallback, Fragment } from 'react';
+import formatMessage from 'format-message';
+import { RouteComponentProps } from '@reach/router';
+import { PublishTarget } from '@bfc/shared';
+import { TextField, ITextFieldProps } from 'office-ui-fabric-react/lib/TextField';
+import { TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
+import { ActionButton, Button } from 'office-ui-fabric-react/lib/Button';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { FontIcon } from 'office-ui-fabric-react/lib/Icon';
+import { Text } from 'office-ui-fabric-react/lib/Text';
+import { Dialog, DialogType } from 'office-ui-fabric-react/lib/Dialog';
+import get from 'lodash/get';
+import { QnAFile, DialogInfo, LuFile } from '@bfc/shared';
+
+import { getBaseName } from '../../utils/fileUtil';
+import { OpenConfirmModal } from '../../components/Modal/ConfirmDialog';
+import { navigateTo } from '../../utils/navigation';
+import {
+  settingsState,
+  localeState,
+  showAddLanguageModalState,
+  publishTypesState,
+  botDisplayNameState,
+  luFilesState,
+  qnaFilesState,
+} from '../../recoilModel/atoms';
+import { rootBotProjectIdSelector } from '../../recoilModel/selectors/project';
+import { dispatcherState, validateDialogSelectorFamily } from '../../recoilModel';
+import { botProjectSpaceSelector } from '../../recoilModel/selectors/project';
+import { CollapsableWrapper } from '../../components/CollapsableWrapper';
+import { AddLanguageModal } from '../../components/MultiLanguage';
+import { languageListTemplates } from '../../components/MultiLanguage';
+import { CreatePublishTarget } from '../publish/createPublishTarget';
+import settingStorage from '../../utils/dialogSettingStorage';
+import ClickOnFocusTextField from '../../components/ClickOnFocusTextField';
+import { isLUISnQnARecognizerType } from '../../utils/dialogValidator';
+
+import { RuntimeSettings } from './runtime-settings/RuntimeSettings';
+import {
+  container,
+  titleStyle,
+  labelContainer,
+  customerLabel,
+  appIdOrPassWordStyle,
+  botLanguageDescriptionStyle,
+  botLanguageFieldStyle,
+  manageBotLanguage,
+  languageItem,
+  botLanguageContainerStyle,
+  customRuntimeStyle,
+  deleteBotText,
+  deleteBotButton,
+  publishTargetsContainer,
+  publishTargetsHeader,
+  publishTargetsHeaderText,
+  publishTargetsItem,
+  publishTargetsItemText,
+  addPublishProfile,
+  editPublishProfile,
+  publishTargetsStyle,
+  publishTargetsEditButton,
+  marginBottom,
+  unknownIconStyle,
+  textFieldStyle,
+  defaultLanguageTextStyle,
+  languageTextStyle,
+} from './styles';
+
+interface BotProjectSettingsProps extends RouteComponentProps<{}> {
+  projectId?: string;
+  hasSkills: boolean;
+}
+
+interface ErrorMsg {
+  MicrosoftAppId?: string;
+  MicrosoftPassword?: string;
+}
+
+const TableView: React.FC<BotProjectSettingsProps> = (props) => {
```

Comment:
give it a better name to represent what it's doing:
e.g., ```BotSettingsTableView```

whatever name you give, please change the file name to that.